audio-worklet/src/noisegate-script-processor.js

```diff
@@ -108,11 +108,11 @@ class NoiseGate {
 
     let envelope = this.detectLevel_(this.channel_);
     envelope = this.convertEnvelopeToSineNormalizedDecibelScale_(envelope);
-
+    let weights = this.computeWeights_(envelope);
```

Comment:
As mentioned before, might as well have `computeWeights_` also do what `convertEnvelopeToSineNormalizedDecibelScale_`, which also allows you to remove this function.